python/ray/train/lightning/lightning_trainer.py

```diff
@@ -0,0 +1,309 @@
+    ``pytorch_lightning.Trainer.fit``.
+
+    Args:
+        lightning_module: A subclass of ``pytorch_lightning.LightningModule``
+            that defines your model and training logic.
+        lightning_module_config: Configurations to pass into
+            ``lightning_module.__init__`` as kwargs.
+        lightning_trainer_config: Configurations to pass into
+            ``pytorch_lightning.Trainer.__init__`` as kwargs. For valid arguments to
+            pass, see
+            https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html#init.
+        lightning_trainer_fit_params: Paramters to pass into
+            ``pytorch_lightning.Trainer.fit``. Specifically, LightningTrainer extracts and 
+            feeds `train_dataloaders`, `val_dataloaders`, and `datamodule` into `Trainer.fit()`. 
+        ddp_strategy_config: Configurations to pass into
+            ``pytorch_lightning.strategies.DDPStrategy.__init__`` as kwargs. Most users
+            should only set this to ``{"find_unused_parameters": False}`` or leave this
+            as-is. For valid arguments to pass, see
+            https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.strategies.DDPStrategy.html
+            and
+            https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html
+        model_checkpoint_config: Configurations used to build a `ModelCheckpoint` callback.
+            The valid arguments list is the same as ``pytorch_lightning.callbacks.ModelCheckpoint``. Please check:
+            https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.callbacks.ModelCheckpoint.html
+        torch_config: Configuration for setting up the PyTorch backend. If set to
+            None, use the default configuration. This replaces the ``backend_config``
+            arg of ``DataParallelTrainer``. Same as in ``TorchTrainer``.
+        scaling_config: Configuration for how to scale data parallel training.
+        dataset_config: Configuration for dataset ingest.
+        run_config: Configuration for the execution of the training run.
+        datasets: A dictionary of Ray Datasets to use for training. 
+            Use the key "train" to denote which dataset is the training
+            dataset and (optionally) key "val" to denote the validation
+            dataset. If a ``preprocessor`` is provided and has not already 
+            been fit, it will be fit on the training dataset. All datasets will be
+            transformed by the ``preprocessor`` if one is provided.
+        datasets_iter_config: Configuration to iterate the provided ray datasets.
+            Please check the valid arguments list here: 
+            :meth:`ray.data.Dataset.iter_torch_batches() <ray.data.Dataset.iter_torch_batches>`.
+        preprocessor: A ray.data.Preprocessor to preprocess the
+            provided datasets.
+        resume_from_checkpoint: A checkpoint to resume training from.
```

Comment:
im worried that theres just way too many configs here....

Reply:
@woshiyyya I wonder if we can group PTL specific parameters in a single config / builder object. that way, we only have to add a single PTLConfig argument to ``__init__`` compared to the base DataParallelTrainer.
```
ptl_config = (
    PTLConfig()
    .trainer_config(...)
    .fit_params(...)
    .ddp_strategy(...)

ptl_trainer = (
    <normal dpt stuff>,
    lighting_config = ptl_config,
)
...
```

Reply:
@gjoliver Yeah that is a great idea. Users can specify the configures they care about, and we'll provide the others with default values.